### ISSUE TITLE
perf(model): retain V4 expert residency across requests

### ISSUE THREAD

**Issue by anthony-chaudhary** [anthony-chaudhary/fak#13479]
# perf(model): retain V4 expert residency across requests

```routing
repo: fak
lane: model
paths: ["internal/model/v4_live_expert.go", "internal/model/v4_expert_runtime.go", "internal/model/weights.go", "internal/model/quant_q4k.go", "internal/model/v4_live_expert_test.go"]
expected_steps: 6
priority: P1
```

## Working spine
Give DeepSeek V4 routed-expert residency the same model/backend lifetime as loaded weights, while request sessions keep only detachable leases.

## Current state
Session.ensureV4LiveExpert constructs newV4ExpertRuntime per session (internal/model/v4_live_expert.go:94-112). That runtime creates a 480 GiB-budget paged ring (v4_expert_runtime.go:45-70). The served request defers Session.Close (internal/agent/inkernel_decode.go:321-332); Session.Close closes v4Expert (internal/model/hal.go:128-130), and runtime.Close frees the ring (v4_expert_runtime.go:73-89). Sequential requests lose every resident routed expert; concurrent requests may duplicate the same expert bytes under separate budgets. The existing SharedExpertRing covers the generic expertRing path, not this v4Expert runtime.

## Why this is next
Expert streaming dominates the memory and I/O budget for the V4 path. A hot cache owned by one request cannot amortize page-ins across users or retain a stable server-wide byte ceiling.

## Parent context
#12952 owns expert eviction policy and warm start; #12640 is the V4.1 native program. This ticket owns the currently reachable DeepSeek V4 runtime lifetime seam only.

## Core through-line
Introduce one bounded V4 runtime owner per (Model, Backend) identity, attach sessions to it, and detach on Session.Close without releasing resident pages. Free the owner once at model weight teardown after all sessions close. Synchronize shared access and preserve the existing 480 GiB or configured single-owner budget.

## Gold-plating boundary
No new expert ranking policy, prefetcher, Engram cache, distributed placement, device kernel, or V4.1 claim.

## Repro witness
From public fak: ```rg -n 'newV4LiveExpert|newPagedRing|freeAll|v4Expert.Close' internal/model/v4_live_expert.go internal/model/v4_expert_runtime.go internal/model/hal.go```. At HEAD ca37a17a, the creation and free calls are under session lifetime.

## Done condition / witness
Two sequential sessions of the same model/backend reuse resident pages; concurrent sessions cannot create duplicate rings; closing one session preserves bytes for another and model teardown frees exactly once. Witness: ```go test -count=1 ./internal/model -run '^TestV4ExpertRuntimeSharedLifetime'```.

## Verifiable Witness
```text
go test -count=1 ./internal/model -run '^TestV4ExpertRuntimeSharedLifetime'
go test -count=1 ./internal/model
```

## Definition of done
- [ ] A counting fake source/backend shows the second sequential session has a cache hit with no repeated expert page-in.
- [ ] Concurrent sessions share one budget, retain distinct session state, and cannot free pages while a peer uses them.
- [ ] Session.Close detaches; CloseWeights with active sessions defers final free until the last detach, then frees the V4 ring exactly once; wrong model/backend identity is refused.
- [ ] Named test executes and owning package tests pass after the last edit.

## File:Line Seam
internal/model/v4_live_expert.go:94-112; v4_expert_runtime.go:45-89; hal.go:128-130; quant_q4k.go:781-798 model teardown.

## Blast radius and fallback
Primary lane model, DeepSeek V4 live expert streaming only. Existing per-session runtime remains the reference while this is open. If shared attachment cannot be admitted, fail closed or retain the current isolated runtime without claiming reuse; never share mutable session logits, KV, or router state.

## Likely files
- internal/model/v4_live_expert.go
- internal/model/v4_expert_runtime.go
- internal/model/weights.go
- internal/model/quant_q4k.go
- internal/model/v4_live_expert_test.go

## Acceptance gate
Counting witness confirms one page-in across sequential sessions, one shared budget under concurrency, and exactly one final free.

## Closure binding
Resolving signed commit cites this issue and carries (fak model).

## Value
- Centrality: Core
- P1 Context: advanced - routed weights persist across native requests
- P2 Net value: advanced - avoids duplicate page-ins and budget multiplication
- P3 Adaptation: preserved - existing session API and isolated fallback
- P4 Operations: advanced - owner lifetime and residency counters are testable

## Work estimate
Estimate: 5 points

## Overall completion contribution
Contribution: 5/8 points

## Completion standard
development
## Witness
The named model regression must execute and count physical or fake page-ins, hits, and frees across sequential and concurrent sessions. Run go test -count=1 ./internal/model -run '^TestV4ExpertRuntimeSharedLifetime' and go test -count=1 ./internal/model.
